### PR TITLE
qa/cephfs: cleanup code for ceph-fuse command

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Plenty redundant and misleading error messages related to FUSE mounts
are printed in output of vstart_runer.py. A small sequence of cleanup should
fix this.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/fuse_mount.py
Comment (joscollin) [inline qa/tasks/cephfs/fuse_mount.py:93]: This fixes the  `mount: /sys/fs/fuse/connections: must be superuser to use mount.` error. 

But there is a new error:

```
2022-08-10 17:26:35,637.637 INFO:tasks.cephfs.fuse_mount:Found client admin socket at /home/jcollin/workspace/ceph/build/asok/client.0.143449.asok
2022-08-10 17:26:35,637.637 DEBUG:__main__:"sudo" was omitted from the following cmd args before execution and logging using function overriding; check vstart_runner.py for more details.
2022-08-10 17:26:35,637.637 DEBUG:__main__:> sudo ./bin/ceph --admin-daemon /home/jcollin/workspace/ceph/build/asok/client.0.143449.asok status
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
admin_socket: exception getting command descriptions: [Errno 13] Permission denied
2022-08-10 17:26:37,737.737 DEBUG:__main__:"sudo" was omitted from the following cmd args before execution and logging using function overriding; check vstart_runner.py for more details.
2022-08-10 17:26:37,737.737 DEBUG:__main__:> sudo ./bin/ceph --admin-daemon /home/jcollin/workspace/ceph/build/asok/client.0.143449.asok status
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
admin_socket: exception getting command descriptions: [Errno 13] Permission denied
```

So I've removed `omit_sudo=False` and the test passed.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:93]: Strange. I did not hit this issue. What test were you running?

Comment (joscollin) [reply qa/tasks/cephfs/fuse_mount.py:93]: I ran the following tests, both error out.

```
 2465  10/08/22 17:24:18 python ../qa/tasks/vstart_runner.py tasks.cephfs.test_mds_metrics.TestMDSMetrics.test_perf_stats_stale_metrics --debug
```
```
 2466  10/08/22 17:26:13  python ../qa/tasks/vstart_runner.py tasks.cephfs.test_mds_metrics.TestMDSMetrics.test_metrics_from_rank --debug
```

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:93]: Runs fine for me -

```
2022-08-11 15:15:22,692.692 INFO:__main__:Stopped test: test_perf_stats_stale_metrics (tasks.cephfs.test_mds_metrics.TestMDSMetrics)
That `ceph fs perf stats` doesn't output stale metrics after the rank0 MDS failover in 66.651653s
2022-08-11 15:15:22,692.692 INFO:__main__:test_perf_stats_stale_metrics (tasks.cephfs.test_mds_metrics.TestMDSMetrics)
2022-08-11 15:15:22,692.692 INFO:__main__:That `ceph fs perf stats` doesn't output stale metrics after the rank0 MDS failover ... ok
2022-08-11 15:15:22,692.692 INFO:__main__:
2022-08-11 15:15:22,693.693 INFO:__main__:----------------------------------------------------------------------
2022-08-11 15:15:22,693.693 INFO:__main__:Ran 1 test in 66.652s
```

File: qa/tasks/vstart_runner.py
Comment (rishabh-d-dave) [inline qa/tasks/vstart_runner.py:699]: `sudo` for `ceph-fuse` command issued by vstart_runner.py is deleted here.

File: qa/tasks/cephfs/fuse_mount.py
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:93]: Inspecting log messages you copied here, seems like ceph-fuse command was executed with superuser privileges which shouldn't be the case because `sudo` will removed by this patch. See https://github.com/ceph/ceph/pull/47533/files#r943310375.

Maybe some commits of this PR were missing it was tested locally?

**Review by joscollin — APPROVED** [review]

File: qa/tasks/cephfs/fuse_mount.py
Comment (vshankar) [inline qa/tasks/cephfs/fuse_mount.py:148]: This looks more than just a _cleanup_. Also, could this result in a hang if the client is blocklisted?

**Comment by rishabh-d-dave** [comment]
jenkins test make check arm64

File: qa/tasks/cephfs/fuse_mount.py
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:148]: True. It'll better to read `/proc/self/mounts` in this case too. I am trying to get this code phrase added as a convenience method - https://github.com/ceph/teuthology/pull/1786. Waiting for review/merge of this PR now.

Comment (vshankar) [reply qa/tasks/cephfs/fuse_mount.py:148]: ack

**Comment by vshankar** [comment]
> jenkins test make check arm64

jenkins tests are failing from last week :/

File: qa/tasks/cephfs/fuse_mount.py
Comment (vshankar) [reply qa/tasks/cephfs/fuse_mount.py:148]: @rishabh-d-dave ping for this and https://github.com/ceph/teuthology/pull/1786

**Comment by vshankar** [comment]
@rishabh-d-dave ping?

**Comment by rishabh-d-dave** [comment]
@vshankar This got pushed down the priority queue since due to fs authorize, fs swap, quota ticket and test_client_cache_size ticket. I'll refresh thid ASAP.

File: qa/tasks/vstart_runner.py
Comment (vshankar) [inline qa/tasks/vstart_runner.py:724]: Shouldn't this be under the `if` block above?

**Comment by vshankar** [comment]
@rishabh-d-dave plans to update this with latest changes.

File: qa/tasks/vstart_runner.py
Comment (rishabh-d-dave) [reply qa/tasks/vstart_runner.py]: Fixed

**Comment by rishabh-d-dave** [comment]
Minor code change.

**Comment by rishabh-d-dave** [comment]
@joscollin Hi, you approved this PR before but I don't know why this wasn't merged. I've rebased and made a minor change. Please approve this PR again if it looks good to you now too. Thanks!

File: qa/tasks/cephfs/fuse_mount.py
Comment (dparmar18) [inline qa/tasks/cephfs/fuse_mount.py]: since `check_status=False` is removed, you're saying that we should abort if we hit an error? Will this have any impact?

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py]: The impact is intended. I've mentioned this in detail in commit message.

File: qa/tasks/cephfs/fuse_mount.py
Comment (dparmar18) [inline qa/tasks/cephfs/fuse_mount.py:91]: we shouldn't use sudo with FUSE right? Do we really need to omit_sudo then?

**Comment by joscollin** [comment]
> @joscollin Hi, you approved this PR before but I don't know why this wasn't merged. I've rebased and made a minor change. Please approve this PR again if it looks good to you now too. Thanks!

I think the changes looks good. But I don't have a working vstart_runner setup to check this. It's broken since long time. Nowadays, I use only shaman for qa.

File: qa/tasks/cephfs/fuse_mount.py
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:91]: Please read commit message for the explanation -

```
Although setting "omit_sudo" to "False" will have no impact on ceph-fuse
mount command during testing with teuthology (because teuthology code
does nothing whether omit_sudo is set or unset), let's set it to False
so that there's no chance for any bugs in future.
```

**Comment by rishabh-d-dave** [comment]
> > @joscollin Hi, you approved this PR before but I don't know why this wasn't merged. I've rebased and made a minor change. Please approve this PR again if it looks good to you now too. Thanks!
> 
> I think the changes looks good. But I don't have a working vstart_runner setup to check this locally. It's broken since long time. Nowadays, I use only shaman for qa.

vstart_runner.py is running fine for me. Let me know if I can help fixing it.

**Comment by vshankar** [comment]
tagging @ceph/cephfs for review.

File: qa/tasks/cephfs/fuse_mount.py
Comment (dparmar18) [inline qa/tasks/cephfs/fuse_mount.py:146]: i think we should remove the `check_status=False` here too @rishabh-d-dave so that we don't try to mount if the module `fuse` addition failed. or do a check with `lsmod | grep module_name`.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:146]: It sounds like a good suggestion but I let's do this on a different PR.

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-09-12_03:57:09-fs-wip-rishabh-2023sep12-testing-default-smithi/7393882

**Comment by rishabh-d-dave** [comment]
Minor fix.

**Review by rishabh-d-dave — COMMENTED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#22-Sep-2023

**Pull request merged**
